src/SimpleParallelAnalyzer.cpp

```diff
@@ -3,160 +3,210 @@
+        // the current state will never be output as a frame. These blocks will detect that case in the available data, and output a frame
+        // immediately, and then another one once the next active edge sample is known.
+        if( !mClock->DoMoreTransitionsExistInCurrentData() )
+        {
+            // There are no transitions available in the current data. Estimate what the frame size might be as 10 if we haven't seen any
+            // frames yet, otherwise use 10% of the last frame size.
+            U64 estimated_frame_size = 10;
+            if( !is_first_frame )
+            {
+                U64 last_frame_sample_count = last_frame.mEndingSampleInclusive - last_frame.mStartingSampleInclusive + 1;
+                estimated_frame_size = last_frame_sample_count * 0.1;
+                if( estimated_frame_size == 0 )
+                {
+                    estimated_frame_size = 1;
+                }
+            }
+
+            // Make sure we haven't gone past the end of the _real_ frame. WouldAdvancingCauseTransition can block, in which case more
+            // transitions might show up.
+            if( mClock->WouldAdvancingCauseTransition( estimated_frame_size ) )
+            {
+                // Move to inactive edge
+                mClock->AdvanceToNextEdge();
+                if( !mClock->DoMoreTransitionsExistInCurrentData() )
+                {
+                    frame.mStartingSampleInclusive = sample;
+                    frame.mEndingSampleInclusive = mClock->GetSampleNumber() - 1;
+                    mResults->AddFrame( frame );
+                    mResults->CommitResults();
+
+                    sample = mClock->GetSampleNumber();
+                }
+            }
+            else
+            {
+                frame.mStartingSampleInclusive = sample;
+                frame.mEndingSampleInclusive = sample + estimated_frame_size - 1;
+                mResults->AddFrame( frame );
+                mResults->CommitResults();
+
+                sample += estimated_frame_size;
+
+                // Move to inactive edge
+                mClock->AdvanceToNextEdge();
+            }
+
+            // Move to active edge
+            mClock->AdvanceToNextEdge();
+        }
+        else
+        {
+            // Move to inactive edge
+            mClock->AdvanceToNextEdge();
+
+            if( !mClock->DoMoreTransitionsExistInCurrentData() )
+            {
+                frame.mStartingSampleInclusive = sample;
+                frame.mEndingSampleInclusive = mClock->GetSampleNumber() - 1;
+                mResults->AddFrame( frame );
+                mResults->CommitResults();
+
+                sample = mClock->GetSampleNumber();
+            }
+
+            // Move to active edge
+            mClock->AdvanceToNextEdge();
+        }
+
+        frame.mStartingSampleInclusive = sample;
+        frame.mEndingSampleInclusive = mClock->GetSampleNumber() - 1;
+        mResults->AddFrame( frame );
```

Comment:
in general, if any of the blocks above this point committed a frame, would this block commit that frame again?

Reply:
This will either:
 1. Commit the a frame for the full period
 1. Commit a second frame for the second half of the period (whether the frames line up with the inactive edge will depend on whether we actually found the inactive edge in the available data)